### DISCUSSION TITLE
Optical interface working for MEP?

### DISCUSSION THREAD

**Discussion by LuederBosse** [ehorvat1/NES-MEP-Reader#6] [General]
HI,
have you tried if the optical interface is working the same as electrical MEP interface?
my MEP is closed with sticker :-( 
Best
Lüder

**Comment by ehorvat1** [comment]
Hi Lüder,
no I did not try the optical interface, but it was my first idea as well since the MEP port is also sealed off by the "provider". I think my provider even prevents reading from optical port .... reason he gave me that a readout on optical port causes a lot of unnecessarry logs which are then transmitted each 24 hours to his main data collectors. 
As I remember some types of Smart meters are sharing the same serial line with MEP port, but I am also not sure if the provided MBK (the encryption key you have to get anyway) would allow this. 
If you are locared in Austria then the provider MUST give you access to the "customer port" which is the MEP port on the Echelon based smart meters. 
Therefore I made a small adapter which connects to the MEP port terminals. So the provider had to come and open the seals/sticker, mount the adapter , re-seal and provide me with a MBK Key for my smartmeter. 
Check out the picture here: https://github.com/ehorvat1/NES-MEP-Reader/blob/main/Hardware/Pictures-Rev1/Assembly.JPG to get an idea about the adapter. The flat ribbon cable fits through the small gaps so the end can be plugged into the main circuit.
BR, Bert
